sql/core/src/main/scala/org/apache/spark/sql/catalyst/util/V2ExpressionBuilder.scala

```diff
@@ -217,6 +217,9 @@ class V2ExpressionBuilder(e: Expression, isPredicate: Boolean = false) {
       generateExpressionWithName("SUBSTRING", children)
     case Upper(child) => generateExpressionWithName("UPPER", Seq(child))
     case Lower(child) => generateExpressionWithName("LOWER", Seq(child))
+    case BitLength(child) => generateExpressionWithName("BIT_LENGTH", Seq(child))
+    case Length(child) => generateExpressionWithName("CHAR_LENGTH", Seq(child))
```

Comment:
we should make sure the chid data type is string type, to safely translate to `CHAR_LENGTH`

Reply:
fixed